server/controllers/medical/ward/bed.js

```diff
@@ -0,0 +1,92 @@
+const db = require('../../../lib/db');
+
+module.exports.create = create;
+module.exports.update = update;
+module.exports.delete = remove;
+module.exports.read = read;
+module.exports.detail = detail;
+
+
+// register a new bed
+function create(req, res, next) {
+  const data = req.body;
+  db.convert(data, ['room_uuid']);
+
+  const sql = 'INSERT INTO bed SET ?';
+  db.exec(sql, data)
+    .then((row) => {
+      res.status(201).json({ id : row.insertId });
+    })
+    .catch(next);
+}
+
+// modify a bed informations
+function update(req, res, next) {
+  const { id } = req.params;
+  let data = req.body;
+
+  if (data.room_uuid) {
+    data = db.convert(data, ['room_uuid']);
+  }
+
+  delete data.id;
+  const sql = `UPDATE bed SET ? WHERE id = ?;`;
+  db.exec(sql, [data, id])
+    .then(() => {
+      res.sendStatus(200);
+    })
+    .catch(next);
+}
+
+// delete a bed
+function remove(req, res, next) {
+  const { id } = req.params;
+  const sql = `DELETE FROM bed WHERE id = ?;`;
+
+  db.exec(sql, [id])
+    .then(() => {
+      res.sendStatus(204);
+    })
+    .catch(next);
+}
+
+// get all beds
+function read(req, res, next) {
```

Comment:
We will eventually want to do more complex queries on these beds (for example, list all beds in the ward X).

Let's make a common function called `lookupBeds(options)` that will use `FilterParser` to filter this beds.  Like this:
```js
function lookupBeds(options) {
  db.convert(options, ['ward_uuid', 'room_uuid']);
  const filters = new FilterParser(options);
  const sql = `
    SELECT b.id, b.label, 
      BUID(r.uuid) as room_uuid, r.label AS room_label,
      BUID(w.uuid) AS ward_uuid, w.name AS ward_name, w.description,
      s.name as service_name
    FROM bed b
    JOIN room r ON r.uuid = b.room_uuid
    JOIN ward w ON w.uuid = r.ward_uuid
    LEFT JOIN service s ON s.id = w.service_id
  `;

   filters.equals('id');
   filters.equals('ward_uuid');
   // etc ...
}
```

That way, our work will be much easier in the future.